From 42ae17e19967e480997d7d5f910bbfe95cfe2017 Mon Sep 17 00:00:00 2001
From: Egon Willighagen <egonw@users.sourceforge.net>
Date: Wed, 11 Nov 2009 21:45:25 +0100
Subject: [PATCH 2/5] Moved all classes into a smsd module.

* added @cdk.module tags
* moved the copyright to above the 'package' statement here and there
---
 .../openscience/cdk/smsd/SubStructureFactory.java  |   41 ++++++++---------
 .../openscience/cdk/smsd/algorithm/cdk/CDKMCS.java |    2 +-
 .../cdk/smsd/algorithm/cdk/CDKMCSHandler.java      |    9 ++--
 .../cdk/smsd/algorithm/cdk/CDKRGraph.java          |    2 +-
 .../cdk/smsd/algorithm/cdk/CDKRMap.java            |    2 +-
 .../cdk/smsd/algorithm/cdk/CDKRMapHandler.java     |   36 +++++++--------
 .../cdk/smsd/algorithm/cdk/CDKRNode.java           |    2 +-
 .../cdk/smsd/algorithm/mcgregor/McGregor.java      |   39 +++++++--------
 .../cdk/smsd/algorithm/mcsplus/BKKCKCF.java        |    2 +
 .../mcsplus/GenerateCompatibilityGraph.java        |   39 +++++++--------
 .../cdk/smsd/algorithm/mcsplus/MCSPlus.java        |   42 ++++++++---------
 .../cdk/smsd/algorithm/mcsplus/MCSPlusHandler.java |   46 +++++++++----------
 .../cdk/smsd/algorithm/single/SingleMapping.java   |   35 ++++++--------
 .../algorithm/single/SingleMappingHandler.java     |   49 ++++++++-----------
 .../cdk/smsd/algorithm/vflib/VFlibHandler.java     |   47 +++++++++----------
 .../cdk/smsd/algorithm/vflib/VFlibMCSHandler.java  |   47 +++++++++----------
 .../smsd/algorithm/vflib/VFlibTurboHandler.java    |   47 +++++++++----------
 .../cdk/smsd/algorithm/vflib/map/VFMCSMapper.java  |   45 ++++++++----------
 .../cdk/smsd/algorithm/vflib/map/VFMapper.java     |   46 ++++++++----------
 .../algorithm/vflib/query/TemplateCompiler.java    |   42 +++++++----------
 .../algorithm/vflib/validator/VFBondMatcher.java   |   31 +++++--------
 .../cdk/smsd/core/canonical/CanonicalLabeler.java  |    2 +-
 .../cdk/smsd/core/canonical/InvPair.java           |    3 +
 .../core/canonical/ReOrganizeAtomContainer.java    |    3 +
 .../openscience/cdk/smsd/core/tools/Distance.java  |    1 +
 .../core/tools/EBIAtomContainerManipulator.java    |    2 +
 .../cdk/smsd/core/tools/EBIBondEnergies.java       |    3 +
 .../cdk/smsd/core/tools/EBIException.java          |    3 +
 .../cdk/smsd/core/tools/EBIMCSCalculator.java      |    3 +
 .../cdk/smsd/core/tools/EBIMDLReader.java          |   47 +++++++++----------
 .../cdk/smsd/core/tools/EBITimeManager.java        |    3 +
 .../cdk/smsd/core/tools/MoleculeSanityCheck.java   |    3 +
 .../cdk/smsd/core/tools/Similarity.java            |    3 +
 .../cdk/smsd/core/tools/SubGraphMapper.java        |    4 +-
 .../openscience/cdk/smsd/factory/MCSFactory.java   |    3 +
 .../cdk/smsd/factory/SubGraphFactory.java          |    3 +
 .../cdk/smsd/filters/ChemicalFilters.java          |    3 +
 .../openscience/cdk/smsd/filters/ExactMapping.java |    3 +
 .../cdk/smsd/filters/FragmentMatcher.java          |    3 +
 .../openscience/cdk/smsd/filters/PostFilter.java   |    3 +
 .../org/openscience/cdk/smsd/global/BondType.java  |    3 +
 .../org/openscience/cdk/smsd/global/TimeOut.java   |    3 +
 .../openscience/cdk/smsd/helper/BinaryTree.java    |    3 +
 .../cdk/smsd/helper/FileExportFilter.java          |   35 ++++++--------
 .../cdk/smsd/helper/FileFilterUtility.java         |   21 ++++-----
 .../openscience/cdk/smsd/helper/FinalMappings.java |    3 +
 .../cdk/smsd/helper/FinderArguments.java           |   30 +++++-------
 .../openscience/cdk/smsd/helper/ImagePreView.java  |   15 ++----
 .../cdk/smsd/helper/LabelContainer.java            |    3 +
 .../openscience/cdk/smsd/helper/MolHandler.java    |    1 +
 .../openscience/cdk/smsd/helper/molFileFilter.java |   21 ++++-----
 .../cdk/smsd/interfaces/IFinalMapping.java         |   25 ++++------
 .../openscience/cdk/smsd/interfaces/IFragment.java |   15 ++----
 .../org/openscience/cdk/smsd/interfaces/IMCS.java  |    3 +-
 .../cdk/smsd/interfaces/IMCSAlgorithm.java         |   27 +++++------
 .../openscience/cdk/smsd/interfaces/IMCSBase.java  |   24 ++++-----
 .../openscience/cdk/smsd/interfaces/ISubGraph.java |   25 ++++------
 57 files changed, 486 insertions(+), 520 deletions(-)

diff --git a/src/main/org/openscience/cdk/smsd/SubStructureFactory.java b/src/main/org/openscience/cdk/smsd/SubStructureFactory.java
index 9da501d..c17886b 100644
--- a/src/main/org/openscience/cdk/smsd/SubStructureFactory.java
+++ b/src/main/org/openscience/cdk/smsd/SubStructureFactory.java
@@ -1,25 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd;
-
-import java.io.IOException;
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
-import org.openscience.cdk.smsd.core.tools.EBIException;
-import org.openscience.cdk.smsd.factory.MCSFactory;
-import org.openscience.cdk.smsd.factory.SubGraphFactory;
-import org.openscience.cdk.smsd.helper.MolHandler;
-import org.openscience.cdk.smsd.interfaces.IMCSAlgorithm;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-import org.openscience.cdk.interfaces.IMolecule;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -41,6 +20,24 @@ import org.openscience.cdk.interfaces.IMolecule;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import org.openscience.cdk.smsd.core.tools.EBIException;
+import org.openscience.cdk.smsd.factory.MCSFactory;
+import org.openscience.cdk.smsd.factory.SubGraphFactory;
+import org.openscience.cdk.smsd.helper.MolHandler;
+import org.openscience.cdk.smsd.interfaces.IMCSAlgorithm;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IMolecule;
+
+/**
+ * @cdk.module smsd
+ */
 public class SubStructureFactory implements IMCSAlgorithm {
 
     IMCSAlgorithm comparison;
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKMCS.java b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKMCS.java
index 3a90409..e1cf9f8 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKMCS.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKMCS.java
@@ -120,7 +120,7 @@ import org.openscience.cdk.tools.manipulator.BondManipulator;
  * @Copyright (C)   2009  Syed Asad Rahman <asad@ebi.ac.uk> (modified the orignal code)
  * @cdk.created 2002-07-17
  * @cdk.require java1.4+
- * @cdk.module  standard
+ * @cdk.module  smsd
  * @cdk.svnrev  $Revision$
  */
 public class CDKMCS {
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKMCSHandler.java b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKMCSHandler.java
index 2ad300f..8d296f4 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKMCSHandler.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKMCSHandler.java
@@ -1,7 +1,4 @@
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -25,7 +22,6 @@
  */
 package org.openscience.cdk.smsd.algorithm.cdk;
 
-//~--- classes ----------------------------------------------------------------
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -43,6 +39,9 @@ import org.openscience.cdk.interfaces.IAtom;
 import org.openscience.cdk.interfaces.IAtomContainer;
 import org.openscience.cdk.interfaces.IMoleculeSet;
 
+/**
+ * @cdk.module smsd
+ */
 public class CDKMCSHandler implements IMCS {
 
 //    //~--- fields -------------------------------------------------------------
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRGraph.java b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRGraph.java
index 896d6a4..338392e 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRGraph.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRGraph.java
@@ -99,7 +99,7 @@ import org.openscience.cdk.smsd.global.TimeOut;
  * @Copyright (C)   2009  Syed Asad Rahman <asad@ebi.ac.uk> (modified the orignal code)
  * @cdk.created 2002-07-17
  * @cdk.require java1.4+
- * @cdk.module  standard
+ * @cdk.module  smsd
  * @cdk.svnrev  $Revision$
  */
 public class CDKRGraph {
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRMap.java b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRMap.java
index 6c29b80..a396f60 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRMap.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRMap.java
@@ -35,7 +35,7 @@ package org.openscience.cdk.smsd.algorithm.cdk;
  * @Copyright (C)   2009  Syed Asad Rahman <asad@ebi.ac.uk> (modified the orignal code)
  * @author      Stephane Werner, IXELIS <mail@ixelis.net>
  * @cdk.created 2002-07-24
- * @cdk.module  standard
+ * @cdk.module  smsd
  * @cdk.svnrev  $Revision$
  */
 public class CDKRMap {
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRMapHandler.java b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRMapHandler.java
index c118883..1e1a49b 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRMapHandler.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRMapHandler.java
@@ -1,22 +1,4 @@
-
-package org.openscience.cdk.smsd.algorithm.cdk;
-
-import java.util.ArrayList;
-import java.util.BitSet;
-import java.util.List;
-import java.util.Stack;
-import java.util.TreeMap;
-import java.util.Vector;
-import org.openscience.cdk.smsd.core.tools.EBIException;
-import org.openscience.cdk.smsd.helper.FinalMappings;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-import org.openscience.cdk.interfaces.IBond;
-import org.openscience.cdk.tools.manipulator.BondManipulator;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -38,6 +20,21 @@ import org.openscience.cdk.tools.manipulator.BondManipulator;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.algorithm.cdk;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.List;
+import java.util.Stack;
+import java.util.TreeMap;
+import java.util.Vector;
+import org.openscience.cdk.smsd.core.tools.EBIException;
+import org.openscience.cdk.smsd.helper.FinalMappings;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IBond;
+import org.openscience.cdk.tools.manipulator.BondManipulator;
+
 /**
  * This algorithm derives from the algorithm described in
  * [Tonnelier, C. and Jauffret, Ph. and Hanser, Th. and Jauffret, Ph. and Kaufmann, G.,
@@ -46,6 +43,7 @@ import org.openscience.cdk.tools.manipulator.BondManipulator;
  * Tetrahedron Comput. Methodol., 1990, 3:351-358] and modified in the thesis of
  * T. Hanser [Unknown BibTeXML type: HAN93].
  *
+ * @cdk.module smsd
  */
 public class CDKRMapHandler {
 
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRNode.java b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRNode.java
index fe86e8b..b979497 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRNode.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/cdk/CDKRNode.java
@@ -43,7 +43,7 @@ import java.util.BitSet;
  *
  * @author      Stephane Werner from IXELIS mail@ixelis.net
  * @cdk.created 2002-07-17
- * @cdk.module  standard
+ * @cdk.module  smsd
  * @cdk.svnrev  $Revision$
  */
 public class CDKRNode {
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/mcgregor/McGregor.java b/src/main/org/openscience/cdk/smsd/algorithm/mcgregor/McGregor.java
index ee76774..a81068e 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/mcgregor/McGregor.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/mcgregor/McGregor.java
@@ -1,24 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.algorithm.mcgregor;
-
-import java.io.IOException;
-import java.util.List;
-import java.util.Map;
-import java.util.Stack;
-import java.util.Vector;
-import org.openscience.cdk.CDKConstants;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-import org.openscience.cdk.interfaces.IBond;
-import org.openscience.cdk.smsd.global.BondType;
-import org.openscience.cdk.smsd.helper.BinaryTree;
-
-/**
- *
- * Copyright (C) 2005-2006 Markus Leber
+/* Copyright (C) 2005-2006 Markus Leber
  *               2006-2009 Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
@@ -41,6 +21,23 @@ import org.openscience.cdk.smsd.helper.BinaryTree;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.algorithm.mcgregor;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+import java.util.Vector;
+import org.openscience.cdk.CDKConstants;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IBond;
+import org.openscience.cdk.smsd.global.BondType;
+import org.openscience.cdk.smsd.helper.BinaryTree;
+
+/**
+ * @cdk.module smsd
+ */
 public class McGregor {
 
     private IAtomContainer ac1 = null;
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/BKKCKCF.java b/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/BKKCKCF.java
index fa2d9d5..07ed7a6 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/BKKCKCF.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/BKKCKCF.java
@@ -38,6 +38,8 @@ import java.util.Vector;
  * described in [F. Cazals, R. Karande: An Algorithm for reporting maximal c-cliques;
  * T.Comp. Sc. (2005); vol 349; pp.
  * 484-490]
+ * 
+ * @cdk.module smsd
  */
 public class BKKCKCF {
 
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/GenerateCompatibilityGraph.java b/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/GenerateCompatibilityGraph.java
index 9b70059..2173314 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/GenerateCompatibilityGraph.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/GenerateCompatibilityGraph.java
@@ -1,24 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.algorithm.mcsplus;
-
-
-import java.io.IOException;
-import java.util.List;
-import java.util.Vector;
-import org.openscience.cdk.smsd.global.BondType;
-import org.openscience.cdk.smsd.helper.LabelContainer;
-import org.openscience.cdk.CDKConstants;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-import org.openscience.cdk.interfaces.IBond;
-import org.openscience.cdk.interfaces.IBond.Order;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -40,6 +20,23 @@ import org.openscience.cdk.interfaces.IBond.Order;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.algorithm.mcsplus;
+
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Vector;
+import org.openscience.cdk.smsd.global.BondType;
+import org.openscience.cdk.smsd.helper.LabelContainer;
+import org.openscience.cdk.CDKConstants;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IBond;
+import org.openscience.cdk.interfaces.IBond.Order;
+
+/**
+ * @cdk.module smsd
+ */
 public class GenerateCompatibilityGraph {
 
     private Vector<Integer> comp_graph_nodes = new Vector<Integer>();
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/MCSPlus.java b/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/MCSPlus.java
index 1925fea..90d4ff8 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/MCSPlus.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/MCSPlus.java
@@ -1,26 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.algorithm.mcsplus;
-
-
-import java.io.IOException;
-import java.util.List;
-import java.util.Stack;
-import java.util.Vector;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-import org.openscience.cdk.smsd.algorithm.mcgregor.McGregor;
-import org.openscience.cdk.smsd.core.tools.EBIException;
-import org.openscience.cdk.smsd.core.tools.EBITimeManager;
-import org.openscience.cdk.smsd.filters.ExactMapping;
-import org.openscience.cdk.smsd.global.TimeOut;
-import org.openscience.cdk.interfaces.IAtomContainer;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -42,6 +20,24 @@ import org.openscience.cdk.interfaces.IAtomContainer;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.algorithm.mcsplus;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Stack;
+import java.util.Vector;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.openscience.cdk.smsd.algorithm.mcgregor.McGregor;
+import org.openscience.cdk.smsd.core.tools.EBIException;
+import org.openscience.cdk.smsd.core.tools.EBITimeManager;
+import org.openscience.cdk.smsd.filters.ExactMapping;
+import org.openscience.cdk.smsd.global.TimeOut;
+import org.openscience.cdk.interfaces.IAtomContainer;
+
+/**
+ * @cdk.module smsd
+ */
 public class MCSPlus {
 
     public static double timeout = TimeOut.getInstance().getTimeOut();
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/MCSPlusHandler.java b/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/MCSPlusHandler.java
index cf283aa..19d7a64 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/MCSPlusHandler.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/mcsplus/MCSPlusHandler.java
@@ -1,5 +1,24 @@
-/*
- * SearchCliques.java
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ * All we ask is that proper credit is given for our work, which includes
+ * - but is not limited to - adding the above copyright notice to the beginning
+ * of your source code files, and to any copyright notice that you may distribute
+ * with programs based on this work.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 package org.openscience.cdk.smsd.algorithm.mcsplus;
 
@@ -21,28 +40,7 @@ import org.openscience.cdk.interfaces.IAtom;
 import org.openscience.cdk.interfaces.IAtomContainer;
 
 /**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
- *
- * Contact: cdk-devel@lists.sourceforge.net
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public License
- * as published by the Free Software Foundation; either version 2.1
- * of the License, or (at your option) any later version.
- * All we ask is that proper credit is given for our work, which includes
- * - but is not limited to - adding the above copyright notice to the beginning
- * of your source code files, and to any copyright notice that you may distribute
- * with programs based on this work.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ * @cdk.module smsd
  */
 public class MCSPlusHandler implements IMCS {
 
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/single/SingleMapping.java b/src/main/org/openscience/cdk/smsd/algorithm/single/SingleMapping.java
index b974ecc..19489a3 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/single/SingleMapping.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/single/SingleMapping.java
@@ -1,22 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.algorithm.single;
-
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
-import java.util.Vector;
-import org.openscience.cdk.smsd.helper.FinalMappings;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-import org.openscience.cdk.interfaces.IBond;
-import org.openscience.cdk.interfaces.IBond.Order;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -38,6 +20,21 @@ import org.openscience.cdk.interfaces.IBond.Order;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.algorithm.single;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.Vector;
+import org.openscience.cdk.smsd.helper.FinalMappings;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IBond;
+import org.openscience.cdk.interfaces.IBond.Order;
+
+/**
+ * @cdk.module smsd
+ */
 public class SingleMapping {
 
     private IAtomContainer ac1 = null;
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/single/SingleMappingHandler.java b/src/main/org/openscience/cdk/smsd/algorithm/single/SingleMappingHandler.java
index 81f57d0..635388a 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/single/SingleMappingHandler.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/single/SingleMappingHandler.java
@@ -1,31 +1,4 @@
-/*
- * @Copyright (C)   2009  Syed Asad Rahman <asad@ebi.ac.uk>
- *
- */
-package org.openscience.cdk.smsd.algorithm.single;
-
-
-
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
-import java.util.Vector;
-import org.openscience.cdk.smsd.core.tools.EBIException;
-import org.openscience.cdk.smsd.helper.FinalMappings;
-import org.openscience.cdk.smsd.helper.MolHandler;
-import org.openscience.cdk.smsd.interfaces.IMCS;
-import org.openscience.cdk.AtomContainer;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-
-
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -47,6 +20,26 @@ import org.openscience.cdk.interfaces.IAtomContainer;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.algorithm.single;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.Vector;
+import org.openscience.cdk.smsd.core.tools.EBIException;
+import org.openscience.cdk.smsd.helper.FinalMappings;
+import org.openscience.cdk.smsd.helper.MolHandler;
+import org.openscience.cdk.smsd.interfaces.IMCS;
+import org.openscience.cdk.AtomContainer;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+
+/**
+ * @cdk.module smsd
+ */
 public class SingleMappingHandler implements IMCS {
 
     private static List<Map<IAtom, IAtom>> allAtomMCS = null;
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibHandler.java b/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibHandler.java
index f0af585..9ca46e7 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibHandler.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibHandler.java
@@ -1,6 +1,24 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ * All we ask is that proper credit is given for our work, which includes
+ * - but is not limited to - adding the above copyright notice to the beginning
+ * of your source code files, and to any copyright notice that you may distribute
+ * with programs based on this work.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 package org.openscience.cdk.smsd.algorithm.vflib;
 import java.io.IOException;
@@ -22,28 +40,7 @@ import org.openscience.cdk.interfaces.IAtom;
 import org.openscience.cdk.interfaces.IAtomContainer;
 
 /**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
- *
- * Contact: cdk-devel@lists.sourceforge.net
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public License
- * as published by the Free Software Foundation; either version 2.1
- * of the License, or (at your option) any later version.
- * All we ask is that proper credit is given for our work, which includes
- * - but is not limited to - adding the above copyright notice to the beginning
- * of your source code files, and to any copyright notice that you may distribute
- * with programs based on this work.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ * @cdk.module smsd
  */
 public class VFlibHandler implements ISubGraph {
 
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibMCSHandler.java b/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibMCSHandler.java
index df15f8a..64145c0 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibMCSHandler.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibMCSHandler.java
@@ -1,6 +1,24 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ * All we ask is that proper credit is given for our work, which includes
+ * - but is not limited to - adding the above copyright notice to the beginning
+ * of your source code files, and to any copyright notice that you may distribute
+ * with programs based on this work.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 package org.openscience.cdk.smsd.algorithm.vflib;
 
@@ -26,28 +44,7 @@ import org.openscience.cdk.interfaces.IAtom;
 import org.openscience.cdk.interfaces.IAtomContainer;
 
 /**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
- *
- * Contact: cdk-devel@lists.sourceforge.net
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public License
- * as published by the Free Software Foundation; either version 2.1
- * of the License, or (at your option) any later version.
- * All we ask is that proper credit is given for our work, which includes
- * - but is not limited to - adding the above copyright notice to the beginning
- * of your source code files, and to any copyright notice that you may distribute
- * with programs based on this work.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ * @cdk.module smsd
  */
 public class VFlibMCSHandler implements IMCS {
 
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibTurboHandler.java b/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibTurboHandler.java
index 07f0e1d..2325218 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibTurboHandler.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/vflib/VFlibTurboHandler.java
@@ -1,6 +1,24 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ * All we ask is that proper credit is given for our work, which includes
+ * - but is not limited to - adding the above copyright notice to the beginning
+ * of your source code files, and to any copyright notice that you may distribute
+ * with programs based on this work.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 package org.openscience.cdk.smsd.algorithm.vflib;
 
@@ -22,28 +40,7 @@ import org.openscience.cdk.interfaces.IAtom;
 import org.openscience.cdk.interfaces.IAtomContainer;
 
 /**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
- *
- * Contact: cdk-devel@lists.sourceforge.net
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public License
- * as published by the Free Software Foundation; either version 2.1
- * of the License, or (at your option) any later version.
- * All we ask is that proper credit is given for our work, which includes
- * - but is not limited to - adding the above copyright notice to the beginning
- * of your source code files, and to any copyright notice that you may distribute
- * with programs based on this work.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ * @cdk.module smsd
  */
 public class VFlibTurboHandler implements ISubGraph {
 
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/vflib/map/VFMCSMapper.java b/src/main/org/openscience/cdk/smsd/algorithm/vflib/map/VFMCSMapper.java
index f7cee02..fb71b4a 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/vflib/map/VFMCSMapper.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/vflib/map/VFMCSMapper.java
@@ -1,28 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.algorithm.vflib.map;
-
-
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IMapper;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.INode;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IQuery;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IState;
-import org.openscience.cdk.smsd.algorithm.vflib.query.TemplateCompiler;
-import org.openscience.cdk.smsd.algorithm.vflib.validator.VFMatch;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-
-
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -44,6 +20,25 @@ import org.openscience.cdk.interfaces.IAtomContainer;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.algorithm.vflib.map;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IMapper;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.INode;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IQuery;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IState;
+import org.openscience.cdk.smsd.algorithm.vflib.query.TemplateCompiler;
+import org.openscience.cdk.smsd.algorithm.vflib.validator.VFMatch;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+
+/**
+ * @cdk.module smsd
+ */
 public class VFMCSMapper implements IMapper {
 
     private IQuery query;
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/vflib/map/VFMapper.java b/src/main/org/openscience/cdk/smsd/algorithm/vflib/map/VFMapper.java
index e414262..9655075 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/vflib/map/VFMapper.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/vflib/map/VFMapper.java
@@ -1,26 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.algorithm.vflib.map;
-
-
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IMapper;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.INode;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IQuery;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IState;
-import org.openscience.cdk.smsd.algorithm.vflib.query.TemplateCompiler;
-import org.openscience.cdk.smsd.algorithm.vflib.validator.VFMatch;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -41,9 +19,7 @@ import org.openscience.cdk.interfaces.IAtomContainer;
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
- */
-
-/*
+ * 
  * MX Cheminformatics Tools for Java
  *
  * Copyright (c) 2007-2009 Metamolecular, LLC
@@ -68,7 +44,25 @@ import org.openscience.cdk.interfaces.IAtomContainer;
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+package org.openscience.cdk.smsd.algorithm.vflib.map;
 
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IMapper;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.INode;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IQuery;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IState;
+import org.openscience.cdk.smsd.algorithm.vflib.query.TemplateCompiler;
+import org.openscience.cdk.smsd.algorithm.vflib.validator.VFMatch;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+
+/**
+ * @cdk.module smsd
+ */
 public class VFMapper implements IMapper {
 
     private IQuery query;
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/vflib/query/TemplateCompiler.java b/src/main/org/openscience/cdk/smsd/algorithm/vflib/query/TemplateCompiler.java
index 62f25f9..0abf8cd 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/vflib/query/TemplateCompiler.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/vflib/query/TemplateCompiler.java
@@ -1,24 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.algorithm.vflib.query;
-
-
-import org.openscience.cdk.smsd.algorithm.vflib.builder.VFQueryBuilder;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IAtomMatcher;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IBondMatcher;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IQuery;
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IQueryCompiler;
-import org.openscience.cdk.smsd.algorithm.vflib.validator.VFAtomMatcher;
-import org.openscience.cdk.smsd.algorithm.vflib.validator.VFBondMatcher;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IAtomContainer;
-import org.openscience.cdk.interfaces.IBond;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -39,9 +19,7 @@ import org.openscience.cdk.interfaces.IBond;
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
- */
-
-/*
+ * 
  * MX Cheminformatics Tools for Java
  *
  * Copyright (c) 2007-2009 Metamolecular, LLC
@@ -66,9 +44,25 @@ import org.openscience.cdk.interfaces.IBond;
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+package org.openscience.cdk.smsd.algorithm.vflib.query;
+
+
+import org.openscience.cdk.smsd.algorithm.vflib.builder.VFQueryBuilder;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IAtomMatcher;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IBondMatcher;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IQuery;
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IQueryCompiler;
+import org.openscience.cdk.smsd.algorithm.vflib.validator.VFAtomMatcher;
+import org.openscience.cdk.smsd.algorithm.vflib.validator.VFBondMatcher;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IBond;
+
 /**
  * @author Richard L. Apodaca <rapodaca at metamolecular.com>
  * @Copyright (C)   2009  Syed Asad Rahman <asad@ebi.ac.uk> (modified the orignal code)
+ * 
+ * @cdk.module smsd
  */
 public class TemplateCompiler implements IQueryCompiler {
 
diff --git a/src/main/org/openscience/cdk/smsd/algorithm/vflib/validator/VFBondMatcher.java b/src/main/org/openscience/cdk/smsd/algorithm/vflib/validator/VFBondMatcher.java
index ee713f4..240461e 100644
--- a/src/main/org/openscience/cdk/smsd/algorithm/vflib/validator/VFBondMatcher.java
+++ b/src/main/org/openscience/cdk/smsd/algorithm/vflib/validator/VFBondMatcher.java
@@ -1,19 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.algorithm.vflib.validator;
-
-
-import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IBondMatcher;
-import org.openscience.cdk.smsd.global.BondType;
-import org.openscience.cdk.CDKConstants;
-import org.openscience.cdk.interfaces.IAtom;
-import org.openscience.cdk.interfaces.IBond;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -34,9 +19,7 @@ import org.openscience.cdk.interfaces.IBond;
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
- */
-
-/*
+ * 
  * MX Cheminformatics Tools for Java
  *
  * Copyright (c) 2007-2009 Metamolecular, LLC
@@ -61,7 +44,17 @@ import org.openscience.cdk.interfaces.IBond;
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+package org.openscience.cdk.smsd.algorithm.vflib.validator;
 
+import org.openscience.cdk.smsd.algorithm.vflib.interfaces.IBondMatcher;
+import org.openscience.cdk.smsd.global.BondType;
+import org.openscience.cdk.CDKConstants;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IBond;
+
+/**
+ * @cdk.module smsd
+ */
 public class VFBondMatcher implements IBondMatcher {
 
     private IBond queryBond;
diff --git a/src/main/org/openscience/cdk/smsd/core/canonical/CanonicalLabeler.java b/src/main/org/openscience/cdk/smsd/core/canonical/CanonicalLabeler.java
index eae4a35..baa482a 100644
--- a/src/main/org/openscience/cdk/smsd/core/canonical/CanonicalLabeler.java
+++ b/src/main/org/openscience/cdk/smsd/core/canonical/CanonicalLabeler.java
@@ -74,7 +74,7 @@ import org.openscience.cdk.interfaces.IAtomContainer;
  * The Collections.sort() method uses a merge sort which is
  * stable and runs in n log(n).
  *
- * @cdk.module standard
+ * @cdk.module smsd
  * @cdk.githash
  *
  * @author   Oliver Horlacher <oliver.horlacher@therastrat.com>
diff --git a/src/main/org/openscience/cdk/smsd/core/canonical/InvPair.java b/src/main/org/openscience/cdk/smsd/core/canonical/InvPair.java
index ea78da6..4e839cd 100644
--- a/src/main/org/openscience/cdk/smsd/core/canonical/InvPair.java
+++ b/src/main/org/openscience/cdk/smsd/core/canonical/InvPair.java
@@ -67,6 +67,9 @@ import org.openscience.cdk.math.Primes;
  * @cdk.module standard
  * @cdk.githash
  */
+/**
+ * @cdk.module smsd
+ */
 public class InvPair implements java.io.Serializable {
 
     private static final long serialVersionUID = -1397634098919863122L;
diff --git a/src/main/org/openscience/cdk/smsd/core/canonical/ReOrganizeAtomContainer.java b/src/main/org/openscience/cdk/smsd/core/canonical/ReOrganizeAtomContainer.java
index e94c1ac..5f75420 100644
--- a/src/main/org/openscience/cdk/smsd/core/canonical/ReOrganizeAtomContainer.java
+++ b/src/main/org/openscience/cdk/smsd/core/canonical/ReOrganizeAtomContainer.java
@@ -44,6 +44,9 @@ import org.openscience.cdk.interfaces.ISingleElectron;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class ReOrganizeAtomContainer {
 
     public final static String INVARIANCE_PAIR = "InvariancePair";
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/Distance.java b/src/main/org/openscience/cdk/smsd/core/tools/Distance.java
index c080a2d..b4095ad 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/Distance.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/Distance.java
@@ -74,6 +74,7 @@ import java.util.BitSet;
  *
  * @ref <B>Willett et.al., Chemical Similarity Searching,</B> <I>J.Chem. Inf. Comput. Sci.</I>, Vol. 38, No. 6, 1998
  *
+ * @cdk.module smsd
  */
 public class Distance {
 
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/EBIAtomContainerManipulator.java b/src/main/org/openscience/cdk/smsd/core/tools/EBIAtomContainerManipulator.java
index 2ba9d2c..3aa81f6 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/EBIAtomContainerManipulator.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/EBIAtomContainerManipulator.java
@@ -73,6 +73,8 @@ import org.w3c.dom.NodeList;
 /**
  * This is an extension of CDK Atom container where we have modfiled few methods
  * Some part of this code was taken from CDK source code and modified
+ * 
+ * @cdk.module smsd
  */
 public class EBIAtomContainerManipulator extends AtomContainerManipulator {
 
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/EBIBondEnergies.java b/src/main/org/openscience/cdk/smsd/core/tools/EBIBondEnergies.java
index 30d9756..6b1207b 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/EBIBondEnergies.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/EBIBondEnergies.java
@@ -34,6 +34,9 @@ import org.openscience.cdk.interfaces.IBond.Order;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class EBIBondEnergies {
 
     private static TreeMap<Integer, List<Object>> bondEngergies = null;
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/EBIException.java b/src/main/org/openscience/cdk/smsd/core/tools/EBIException.java
index b3b98ca..d4633db 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/EBIException.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/EBIException.java
@@ -25,6 +25,9 @@ package org.openscience.cdk.smsd.core.tools;
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+/**
+ * @cdk.module smsd
+ */
 public class EBIException extends Exception {
 
     private static final long serialVersionUID = 8371328769230823678L;
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/EBIMCSCalculator.java b/src/main/org/openscience/cdk/smsd/core/tools/EBIMCSCalculator.java
index c71ed5b..63c4c12 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/EBIMCSCalculator.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/EBIMCSCalculator.java
@@ -29,6 +29,9 @@ import org.openscience.cdk.smsd.SubStructureFactory;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class EBIMCSCalculator extends SubStructureFactory {
 
 
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/EBIMDLReader.java b/src/main/org/openscience/cdk/smsd/core/tools/EBIMDLReader.java
index 7a1f221..975ccfe 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/EBIMDLReader.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/EBIMDLReader.java
@@ -1,6 +1,24 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ * All we ask is that proper credit is given for our work, which includes
+ * - but is not limited to - adding the above copyright notice to the beginning
+ * of your source code files, and to any copyright notice that you may distribute
+ * with programs based on this work.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 package org.openscience.cdk.smsd.core.tools;
 
@@ -25,28 +43,7 @@ import org.openscience.cdk.layout.StructureDiagramGenerator;
 import org.openscience.cdk.tools.manipulator.MoleculeSetManipulator;
 
 /**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
- *
- * Contact: cdk-devel@lists.sourceforge.net
- *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public License
- * as published by the Free Software Foundation; either version 2.1
- * of the License, or (at your option) any later version.
- * All we ask is that proper credit is given for our work, which includes
- * - but is not limited to - adding the above copyright notice to the beginning
- * of your source code files, and to any copyright notice that you may distribute
- * with programs based on this work.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ * @cdk.module smsd
  */
 public class EBIMDLReader {
 
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/EBITimeManager.java b/src/main/org/openscience/cdk/smsd/core/tools/EBITimeManager.java
index 876fcde..f9c0623 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/EBITimeManager.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/EBITimeManager.java
@@ -31,6 +31,9 @@ import java.util.TimeZone;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class EBITimeManager {
 
     /**
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/MoleculeSanityCheck.java b/src/main/org/openscience/cdk/smsd/core/tools/MoleculeSanityCheck.java
index 0a8fb13..6354cde 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/MoleculeSanityCheck.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/MoleculeSanityCheck.java
@@ -39,6 +39,9 @@ import org.openscience.cdk.tools.manipulator.RingSetManipulator;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class MoleculeSanityCheck {
 
     /**
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/Similarity.java b/src/main/org/openscience/cdk/smsd/core/tools/Similarity.java
index ad6b923..6e8eeb4 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/Similarity.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/Similarity.java
@@ -75,6 +75,9 @@ import java.util.BitSet;
  *
  *
  */
+/**
+ * @cdk.module smsd
+ */
 public class Similarity {
 
     private static BitSet A;
diff --git a/src/main/org/openscience/cdk/smsd/core/tools/SubGraphMapper.java b/src/main/org/openscience/cdk/smsd/core/tools/SubGraphMapper.java
index fdceb7a..b08ae03 100644
--- a/src/main/org/openscience/cdk/smsd/core/tools/SubGraphMapper.java
+++ b/src/main/org/openscience/cdk/smsd/core/tools/SubGraphMapper.java
@@ -39,7 +39,9 @@ import org.openscience.cdk.interfaces.IBond;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
-
+/**
+ * @cdk.module smsd
+ */
 public class SubGraphMapper {
 
     /**
diff --git a/src/main/org/openscience/cdk/smsd/factory/MCSFactory.java b/src/main/org/openscience/cdk/smsd/factory/MCSFactory.java
index 85ea7c4..2f79bac 100644
--- a/src/main/org/openscience/cdk/smsd/factory/MCSFactory.java
+++ b/src/main/org/openscience/cdk/smsd/factory/MCSFactory.java
@@ -58,6 +58,9 @@ import org.openscience.cdk.interfaces.IMolecule;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class MCSFactory implements IMCSAlgorithm {
 
     private List<TreeMap<Integer, Integer>> allMCS = null;
diff --git a/src/main/org/openscience/cdk/smsd/factory/SubGraphFactory.java b/src/main/org/openscience/cdk/smsd/factory/SubGraphFactory.java
index ee6c386..2c200dc 100644
--- a/src/main/org/openscience/cdk/smsd/factory/SubGraphFactory.java
+++ b/src/main/org/openscience/cdk/smsd/factory/SubGraphFactory.java
@@ -51,6 +51,9 @@ import org.openscience.cdk.interfaces.IMolecule;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class SubGraphFactory implements IMCSAlgorithm {
 
     private List<TreeMap<Integer, Integer>> allMCS = null;
diff --git a/src/main/org/openscience/cdk/smsd/filters/ChemicalFilters.java b/src/main/org/openscience/cdk/smsd/filters/ChemicalFilters.java
index ce1836f..4507cd0 100644
--- a/src/main/org/openscience/cdk/smsd/filters/ChemicalFilters.java
+++ b/src/main/org/openscience/cdk/smsd/filters/ChemicalFilters.java
@@ -55,6 +55,9 @@ import org.openscience.cdk.tools.manipulator.RingSetManipulator;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class ChemicalFilters {
 
     private List<TreeMap<Integer, Integer>> allMCS = null;
diff --git a/src/main/org/openscience/cdk/smsd/filters/ExactMapping.java b/src/main/org/openscience/cdk/smsd/filters/ExactMapping.java
index c5aa6e1..3629e17 100644
--- a/src/main/org/openscience/cdk/smsd/filters/ExactMapping.java
+++ b/src/main/org/openscience/cdk/smsd/filters/ExactMapping.java
@@ -31,6 +31,9 @@ import java.util.Vector;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class ExactMapping {
 
     //extract atom mapping from the clique vector and store it in vector clique_MAPPING
diff --git a/src/main/org/openscience/cdk/smsd/filters/FragmentMatcher.java b/src/main/org/openscience/cdk/smsd/filters/FragmentMatcher.java
index a44dc71..8b02927 100644
--- a/src/main/org/openscience/cdk/smsd/filters/FragmentMatcher.java
+++ b/src/main/org/openscience/cdk/smsd/filters/FragmentMatcher.java
@@ -53,6 +53,9 @@ import org.openscience.cdk.interfaces.IAtomContainerSet;
  *
  * @Copyright (C)   2009  Syed Asad Rahman <asad@ebi.ac.uk>
  */
+/**
+ * @cdk.module smsd
+ */
 public class FragmentMatcher implements IFragment {
 
     // private boolean[][] localFlagMatrix;
diff --git a/src/main/org/openscience/cdk/smsd/filters/PostFilter.java b/src/main/org/openscience/cdk/smsd/filters/PostFilter.java
index 418be03..898082f 100644
--- a/src/main/org/openscience/cdk/smsd/filters/PostFilter.java
+++ b/src/main/org/openscience/cdk/smsd/filters/PostFilter.java
@@ -31,6 +31,9 @@ import java.util.Vector;
 import org.openscience.cdk.smsd.helper.FinalMappings;
 
 
+/**
+ * @cdk.module smsd
+ */
 public class PostFilter {
 
     /** 
diff --git a/src/main/org/openscience/cdk/smsd/global/BondType.java b/src/main/org/openscience/cdk/smsd/global/BondType.java
index a2dcfc4..5142719 100644
--- a/src/main/org/openscience/cdk/smsd/global/BondType.java
+++ b/src/main/org/openscience/cdk/smsd/global/BondType.java
@@ -27,6 +27,9 @@ package org.openscience.cdk.smsd.global;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class BondType {
 
     private static BondType INSTANCE = null;
diff --git a/src/main/org/openscience/cdk/smsd/global/TimeOut.java b/src/main/org/openscience/cdk/smsd/global/TimeOut.java
index c38a8e8..1e8cdf3 100644
--- a/src/main/org/openscience/cdk/smsd/global/TimeOut.java
+++ b/src/main/org/openscience/cdk/smsd/global/TimeOut.java
@@ -28,6 +28,9 @@ package org.openscience.cdk.smsd.global;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/**
+ * @cdk.module smsd
+ */
 public class TimeOut {
 
     private static TimeOut INSTANCE = null;
diff --git a/src/main/org/openscience/cdk/smsd/helper/BinaryTree.java b/src/main/org/openscience/cdk/smsd/helper/BinaryTree.java
index 1218413..72eb8f9 100644
--- a/src/main/org/openscience/cdk/smsd/helper/BinaryTree.java
+++ b/src/main/org/openscience/cdk/smsd/helper/BinaryTree.java
@@ -27,6 +27,9 @@ package org.openscience.cdk.smsd.helper;
 
 //the second part of the program extents the mapping by the McGregor algorithm in case
 //that not all atoms of molecule A and molecule B are mapped by the clique approach
+/**
+ * @cdk.module smsd
+ */
 public class BinaryTree {
 
     /** Creates a new instance of BinaryTree
diff --git a/src/main/org/openscience/cdk/smsd/helper/FileExportFilter.java b/src/main/org/openscience/cdk/smsd/helper/FileExportFilter.java
index 5aa24b6..326c3ea 100644
--- a/src/main/org/openscience/cdk/smsd/helper/FileExportFilter.java
+++ b/src/main/org/openscience/cdk/smsd/helper/FileExportFilter.java
@@ -1,22 +1,4 @@
-package org.openscience.cdk.smsd.helper;
-
-import java.io.File;
-import java.util.List;
-import java.util.ArrayList;
-
-import javax.swing.JFileChooser;
-
-/**
- * An export filter for JCP file formats
- *
- * @cdk.module jchempaint 
- * @author  Egon Willighagen
- * @cdk.created 2003-04-01
- */
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -38,6 +20,21 @@ import javax.swing.JFileChooser;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.helper;
+
+import java.io.File;
+import java.util.List;
+import java.util.ArrayList;
+
+import javax.swing.JFileChooser;
+
+/**
+ * An export filter for JCP file formats
+ *
+ * @cdk.module  smsd
+ * @author      Egon Willighagen
+ * @cdk.created 2003-04-01
+ */
 public class FileExportFilter extends javax.swing.filechooser.FileFilter{
 
     // only those extensions are given here that are *not* on JCPFileFilter
diff --git a/src/main/org/openscience/cdk/smsd/helper/FileFilterUtility.java b/src/main/org/openscience/cdk/smsd/helper/FileFilterUtility.java
index 13425e8..1827e8e 100644
--- a/src/main/org/openscience/cdk/smsd/helper/FileFilterUtility.java
+++ b/src/main/org/openscience/cdk/smsd/helper/FileFilterUtility.java
@@ -1,15 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.helper;
-
-import java.io.File;
-import javax.swing.ImageIcon;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -31,6 +20,14 @@ import javax.swing.ImageIcon;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.helper;
+
+import java.io.File;
+import javax.swing.ImageIcon;
+
+/**
+ * @cdk.module smsd
+ */
 public class FileFilterUtility {
 
 //    public final static String jpeg = "jpeg";
diff --git a/src/main/org/openscience/cdk/smsd/helper/FinalMappings.java b/src/main/org/openscience/cdk/smsd/helper/FinalMappings.java
index 8374577..f385f52 100644
--- a/src/main/org/openscience/cdk/smsd/helper/FinalMappings.java
+++ b/src/main/org/openscience/cdk/smsd/helper/FinalMappings.java
@@ -30,6 +30,9 @@ import java.util.List;
 import java.util.TreeMap;
 import java.util.Vector;
 
+/**
+ * @cdk.module smsd
+ */
 public class FinalMappings implements IFinalMapping {
 
     private List<TreeMap<Integer, Integer>> _mappings;
diff --git a/src/main/org/openscience/cdk/smsd/helper/FinderArguments.java b/src/main/org/openscience/cdk/smsd/helper/FinderArguments.java
index 851669c..35684af 100644
--- a/src/main/org/openscience/cdk/smsd/helper/FinderArguments.java
+++ b/src/main/org/openscience/cdk/smsd/helper/FinderArguments.java
@@ -1,20 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.helper;
-
-import java.util.ArrayList;
-import java.util.List;
-import org.apache.commons.cli.CommandLine;
-import org.apache.commons.cli.CommandLineParser;
-import org.apache.commons.cli.HelpFormatter;
-import org.apache.commons.cli.Options;
-import org.apache.commons.cli.PosixParser;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -36,7 +20,19 @@ import org.apache.commons.cli.PosixParser;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.helper;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.PosixParser;
 
+/**
+ * @cdk.module smsd
+ */
 public class FinderArguments {
 
     //define the size of the fingerprint
diff --git a/src/main/org/openscience/cdk/smsd/helper/ImagePreView.java b/src/main/org/openscience/cdk/smsd/helper/ImagePreView.java
index 195b068..1e9dc09 100644
--- a/src/main/org/openscience/cdk/smsd/helper/ImagePreView.java
+++ b/src/main/org/openscience/cdk/smsd/helper/ImagePreView.java
@@ -1,13 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-
-package org.openscience.cdk.smsd.helper;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -29,6 +20,7 @@ package org.openscience.cdk.smsd.helper;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.helper;
 
 import javax.swing.*;
 import java.beans.*;
@@ -36,6 +28,9 @@ import java.awt.*;
 import java.io.File;
 
 /* ImagePreview.java by FileChooserDemo2.java. */
+/**
+ * @cdk.module smsd
+ */
 public class ImagePreView extends JComponent
                           implements PropertyChangeListener {
     ImageIcon thumbnail = null;
diff --git a/src/main/org/openscience/cdk/smsd/helper/LabelContainer.java b/src/main/org/openscience/cdk/smsd/helper/LabelContainer.java
index 363dd6f..a062c30 100644
--- a/src/main/org/openscience/cdk/smsd/helper/LabelContainer.java
+++ b/src/main/org/openscience/cdk/smsd/helper/LabelContainer.java
@@ -27,6 +27,9 @@ package org.openscience.cdk.smsd.helper;
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * @cdk.module smsd
+ */
 public class LabelContainer {
 
     private HashMap<String, Integer> _labelMap;
diff --git a/src/main/org/openscience/cdk/smsd/helper/MolHandler.java b/src/main/org/openscience/cdk/smsd/helper/MolHandler.java
index f70147e..2820939 100644
--- a/src/main/org/openscience/cdk/smsd/helper/MolHandler.java
+++ b/src/main/org/openscience/cdk/smsd/helper/MolHandler.java
@@ -49,6 +49,7 @@ import org.openscience.cdk.io.MDLReader;
  *
  * @author Syed Asad Rahman, EMBL-EBI, Cambridge, UK
  * @contact asad@ebi.ac.uk
+ * @cdk.module smsd
  */
 public class MolHandler {
 
diff --git a/src/main/org/openscience/cdk/smsd/helper/molFileFilter.java b/src/main/org/openscience/cdk/smsd/helper/molFileFilter.java
index ee29e0c..cece610 100644
--- a/src/main/org/openscience/cdk/smsd/helper/molFileFilter.java
+++ b/src/main/org/openscience/cdk/smsd/helper/molFileFilter.java
@@ -1,15 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.helper;
-
-import java.io.File;
-import javax.swing.filechooser.FileFilter;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -31,6 +20,14 @@ import javax.swing.filechooser.FileFilter;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.helper;
+
+import java.io.File;
+import javax.swing.filechooser.FileFilter;
+
+/**
+ * @cdk.module smsd
+ */
 public class molFileFilter extends FileFilter {
     //Accept all directories and all gif, jpg, tiff, or png files.
 
diff --git a/src/main/org/openscience/cdk/smsd/interfaces/IFinalMapping.java b/src/main/org/openscience/cdk/smsd/interfaces/IFinalMapping.java
index f82a154..91663f4 100644
--- a/src/main/org/openscience/cdk/smsd/interfaces/IFinalMapping.java
+++ b/src/main/org/openscience/cdk/smsd/interfaces/IFinalMapping.java
@@ -1,18 +1,4 @@
-/*
- * IFinalMapping.java
- *
- * Created on Sep 12, 2007, 8:58:45 PM
- *
- */
-package org.openscience.cdk.smsd.interfaces;
-
-import java.util.Iterator;
-import java.util.List;
-import java.util.TreeMap;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -34,6 +20,15 @@ import java.util.TreeMap;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.interfaces;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.TreeMap;
+
+/**
+ * @cdk.module smsd
+ */
 public interface IFinalMapping {
 
    
diff --git a/src/main/org/openscience/cdk/smsd/interfaces/IFragment.java b/src/main/org/openscience/cdk/smsd/interfaces/IFragment.java
index 5d280b1..04a1e82 100644
--- a/src/main/org/openscience/cdk/smsd/interfaces/IFragment.java
+++ b/src/main/org/openscience/cdk/smsd/interfaces/IFragment.java
@@ -1,12 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.interfaces;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -28,6 +20,11 @@ package org.openscience.cdk.smsd.interfaces;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.interfaces;
+
+/**
+ * @cdk.module smsd
+ */
 public interface IFragment extends IMCSBase {
 
     boolean getFlag();
diff --git a/src/main/org/openscience/cdk/smsd/interfaces/IMCS.java b/src/main/org/openscience/cdk/smsd/interfaces/IMCS.java
index 5b7e04b..090c990 100644
--- a/src/main/org/openscience/cdk/smsd/interfaces/IMCS.java
+++ b/src/main/org/openscience/cdk/smsd/interfaces/IMCS.java
@@ -32,8 +32,7 @@ import org.openscience.cdk.smsd.helper.MolHandler;
 import org.openscience.cdk.interfaces.IAtomContainer;
 
 /**
- *
- * @Copyright (C)   2009  Syed Asad Rahman <asad@ebi.ac.uk>
+ * @cdk.module smsd
  */
 public interface IMCS extends IMCSBase {
 
diff --git a/src/main/org/openscience/cdk/smsd/interfaces/IMCSAlgorithm.java b/src/main/org/openscience/cdk/smsd/interfaces/IMCSAlgorithm.java
index 1a1d6c5..7b8f5a1 100644
--- a/src/main/org/openscience/cdk/smsd/interfaces/IMCSAlgorithm.java
+++ b/src/main/org/openscience/cdk/smsd/interfaces/IMCSAlgorithm.java
@@ -1,18 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.interfaces;
-
-import java.io.IOException;
-import org.openscience.cdk.smsd.core.tools.EBIException;
-import org.openscience.cdk.smsd.helper.MolHandler;
-import org.openscience.cdk.interfaces.IAtomContainer;
-import org.openscience.cdk.interfaces.IMolecule;
-
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -34,6 +20,17 @@ import org.openscience.cdk.interfaces.IMolecule;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.interfaces;
+
+import java.io.IOException;
+import org.openscience.cdk.smsd.core.tools.EBIException;
+import org.openscience.cdk.smsd.helper.MolHandler;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IMolecule;
+
+/**
+ * @cdk.module smsd
+ */
 public interface IMCSAlgorithm extends IMCSBase {
 
     Double getEnergyScore(int Key);
diff --git a/src/main/org/openscience/cdk/smsd/interfaces/IMCSBase.java b/src/main/org/openscience/cdk/smsd/interfaces/IMCSBase.java
index 3f69f3f..2f786e9 100644
--- a/src/main/org/openscience/cdk/smsd/interfaces/IMCSBase.java
+++ b/src/main/org/openscience/cdk/smsd/interfaces/IMCSBase.java
@@ -1,16 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.interfaces;
-
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
-import org.openscience.cdk.interfaces.IAtom;
-/**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -32,6 +20,16 @@ import org.openscience.cdk.interfaces.IAtom;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.interfaces;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import org.openscience.cdk.interfaces.IAtom;
+
+/**
+ * @cdk.module smsd
+ */
 public interface IMCSBase {
 
     /**
diff --git a/src/main/org/openscience/cdk/smsd/interfaces/ISubGraph.java b/src/main/org/openscience/cdk/smsd/interfaces/ISubGraph.java
index 8786fac..4572427 100644
--- a/src/main/org/openscience/cdk/smsd/interfaces/ISubGraph.java
+++ b/src/main/org/openscience/cdk/smsd/interfaces/ISubGraph.java
@@ -1,17 +1,4 @@
-/*
- * To change this template, choose Tools | Templates
- * and open the template in the editor.
- */
-package org.openscience.cdk.smsd.interfaces;
-
-import java.io.IOException;
-import org.openscience.cdk.smsd.core.tools.EBIException;
-import org.openscience.cdk.smsd.helper.MolHandler;
-import org.openscience.cdk.interfaces.IAtomContainer;
-
- /**
- *
- * Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
+/* Copyright (C) 2006-2009  Syed Asad Rahman {asad@ebi.ac.uk}
  *
  * Contact: cdk-devel@lists.sourceforge.net
  *
@@ -33,6 +20,16 @@ import org.openscience.cdk.interfaces.IAtomContainer;
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+package org.openscience.cdk.smsd.interfaces;
+
+import java.io.IOException;
+import org.openscience.cdk.smsd.core.tools.EBIException;
+import org.openscience.cdk.smsd.helper.MolHandler;
+import org.openscience.cdk.interfaces.IAtomContainer;
+
+/**
+ * @cdk.module smsd
+ */
 public interface ISubGraph extends IMCSBase {
 
     /**
-- 
1.6.3.3

